src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala

```diff
@@ -195,14 +189,16 @@ trait ChoosesMode {
                 filterStreetVehiclesForQuery(streetVehicles, mode).headOption
               maybeVehicle match {
                 case Some(vehicle) =>
+                  val linkIds = new ArrayBuffer[Int](2 + r.getLinkIds.size())
+                  linkIds += r.getStartLinkId.toString.toInt
+                  r.getLinkIds.asScala.foreach { id => linkIds += id.toString.toInt }
+                  linkIds += r.getStartLinkId.toString.toInt
```

Comment:
I would rather follow scala approach. But not sure about performance in this case.
```Scala
val linkIdList = List(r.getStartLinkId) 
        ::: r.getLinkIds.asScala.toList
        ::: List(r.getStartLinkId)
val linkIds = linkIdList.map(_.toString.toInt)
```

Reply:
Yah...the more functional approach comes at an allocation cost, which is what is being tackled here.